### PR TITLE
Respect $fish_history environment variable

### PR DESCRIPTION
**Pull request by hupfdule**
If `$fish_history` is set, use it to construct the name of the fish
history file.
The behaviour of `$fish_history` is described in
https://fishshell.com/docs/current/cmds/history.html#customizing-the-name-of-the-history-file.

Additionally the actual error is printed if opening the history file failed.

### CODE REVIEW

File: re-search.c [hunk 1]
```diff
@@ -183,13 +186,27 @@ int parse_history() {
 #endif
 
 #ifdef BASH
-	fp = try_open_history(".bash_history");
+	histpath = "."
```
Comment (jbonjean) [inline re-search.c:189]: Missing semicolon.

File: re-search.c [hunk 2]
```diff
@@ -183,13 +186,27 @@ int parse_history() {
...
+	histfile = ".bash_history";
```
Comment (jbonjean) [inline re-search.c:190]: You cannot assign a char array like that.

File: re-search.c [hunk 3]
```diff
@@ -183,13 +186,27 @@ int parse_history() {
...
+	snprintf(histfile, sizeof(histfile), ".bash_history", fish_session);
```
Comment (jbonjean) [inline re-search.c:191]: fish_session is not defined here.

**Review by jbonjean — CHANGES_REQUESTED** [review]
I started the review, but I think this needs to be reworked, at least the bash support, which I think didn't actually need any change.

**Comment by hupfdule** [comment]
You are right, I forgot to correctly adapt the bash history loading and left it in a broken state.

As I decided to maintain my own fork, I will close this PR as well.

**Pull request closed without merging**